### ISSUE TITLE
Experimental Gutenberg feature may have been removed, needs investigation

### ISSUE THREAD

**Issue by deBhal** [Automattic/wp-calypso#48585]
The use of `__experimentalMainDashboardButton` of `@wordpress/edit-site` in `apps/wpcom-block-editor/src/calypso/features/iframe-bridge-server.js` needs a look by someone who understands what it's for to see if it's working and/or obsolete.

#### Steps to reproduce

I found this entry property in https://github.com/Automattic/wp-calypso/pull/48164 where it was listed as an experimental feature we use, but the test is failing.

The test checks whether `wp.editSite.__experimentalMainDashboardButton` is available in the global scope within the block editor and reports that `wp.editSite` is missing entirely, which I've found to be true in manual checks.

However, this doesn't match our use of the property. In `apps/wpcom-block-editor/src/calypso/features/iframe-bridge-server.js`[#](apps/wpcom-block-editor/src/calypso/features/iframe-bridge-server.js) We `const editSitePackage = require( '@wordpress/edit-site' );` and then check whether it exists later, so it's possible that I'm just not testing it properly.

However, when I log `editSitePackage` it always seems to be `null` in both simple and atomic sites. Perhaps it's intended for .org sites with older versions of gutenberg?

On the other hand, it might be related to `yarn`s package magic - the package does not appear in the `node_modules` of `apps/wpcom-block-editor`, only under the root calypso `node_modules`.

Either way, `@wordpress/edit-site` has a number of requirements for it's environment, but I got it to load in `node` and it didn't have the property anyway:
```javascript
jsdom = require('jsdom'); dom = new jsdom.JSDOM(); global.window = dom.window; global.document = window.document; 
window.matchMedia = () => ({addListener: () => ({})})
navigator = {userAgent: '' }
editSitePackage = require( '@wordpress/edit-site' ); // This is what we use in server-interface-bridge.js
Object.keys( editSitePackage )
require('@wordpress/edit-site/package.json').version
```

Currently, the version is `v1.15.6` and the package keys are `[ 'initialize', '__experimentalFullscreenModeClose', '__experimentalNavigationToggle' ]`.

This is confirmed by `grep`ping through the module code - `__experimentalMainDashboardButton` is in `@wordpress/interface`, but not `@wordpress/edit-site`.

To add another layer of uncertainty, the property _does_ appear in a later version. At time of writing, the version in the `gutenberg/master` repo/branch is 1.16 and the package does have the experimental property.

If we run `node` in the gutenberg repo:

```js
editSiteRoot = './packages/edit-site/'
require( editSiteRoot + 'package.json' ).version // '1.16.0'
require( editSiteRoot + 'package.json' ).main    // 'build/index.js'
// hack up an an environment
jsdom = require('jsdom'); dom = new jsdom.JSDOM(); global.window = dom.window; global.document = window.document; 
window.matchMedia = () => ({addListener: () => ({})})
navigator = {userAgent: '' }
editSitePackage = require( editSiteRoot + require( editSiteRoot + 'package.json' ).main ) // that's './packages/edit-site/build/index.js'
// Tada:
typeof editSitePackage.__experimentalMainDashboardButton // 'function'
```

**Comment by sirreal** [comment]
It appears to be exported from two packages, `edit-site` and `edit-post`. The test isn't ideal in that it loads a post so relies on whatever scripts are loaded and cannot change the script dependencies, but `edit-site` will never be loaded in the page the test currently uses.

https://github.com/Automattic/wp-calypso/blob/5a93887e62071a872c9f5b05b55828c8c940863f/test/e2e/specs/wp-gutenberg-experimental-features-spec.js#L29

If this line were `@wordpress/edit-post` instead of `@wordpress/edit-site` I believe the test would pass:

```js
typeof wp.editPost.__experimentalMainDashboardButton
// "function"
```

**Comment by deBhal** [comment]
Does `iframe-bridge-server` need the same `s/edit-site/edit-post` or should we remove both the test and the code that's currently checking the wrong package instead?

I believe the test failure is telling us that this "back to dashboard fill" `render()` will always be null:

https://github.com/Automattic/wp-calypso/blob/7c89b922988fa7d57ad71f2e83fdbc442cc10818/apps/wpcom-block-editor/src/calypso/features/iframe-bridge-server.js#L564-L570

We should do the same thing to the test and the code, but I think we've got a valid choice between "fix" and "remove" depending on whether we still want that back button.

**Comment by sirreal** [comment]
Ok, this took me a bit to figure out 🙂 

Yes, I believe both of these should be changed.

What's happening right now is that we're bundling the @wordpress/interface package. That will be caused by this line: https://github.com/WordPress/gutenberg/blob/541141d49bd48da25b22f568506cca1f24ab5b5b/packages/dependency-extraction-webpack-plugin/lib/util.js#L2

It means that the interface package isn't exposed as a WordPress script, so consumers include it completely within their bundles instead of using the "external" script that is provided by the WordPress environment.

That's the reason these features are still working instead of being broken because the feature is gone/moved.

Calypso happens to be on an older version of the published packages, so the bundled interface package does include the experimental export.

@jsnajdr is already taking care of fixing the imports in https://github.com/Automattic/wp-calypso/pull/48729

**Comment by deBhal** [comment]
Thanks for looking into it! I've added the "Fixes: #48585" to the description there so this gets closed automatically.
